src/inference/src/dev/icompiled_model.cpp

```diff
@@ -48,34 +48,47 @@ ov::ICompiledModel::ICompiledModel(const std::shared_ptr<const ov::Model>& model
             }
         }
 
-        if (add_operation_names) {
-            for (const auto& param : model->get_parameters()) {
-                const auto& param_name = param->get_friendly_name();
+        for (const auto& param : model->get_parameters()) {
+            const auto& param_name = param->get_friendly_name();
+            auto new_param = ov::as_type_ptr<ov::op::v0::Parameter>(param->copy_with_new_inputs({}));
+            new_param->set_friendly_name(param_name);
+            if (add_operation_names) {
                 OPENVINO_ASSERT(!m_plugin->is_new_api() || leaf_names.find(param_name) == leaf_names.end() ||
                                     param->output(0).get_names().find(param_name) != param->output(0).get_names().end(),
                                 "Model operation names have collisions with tensor names.",
                                 " Please use MO to generate new IR version, it should allow to avoid the issue");
                 leaf_names.insert(param_name);
-                param->output(0).get_tensor().add_names({param_name});
-                m_inputs.emplace_back(
-                    ov::Output<const ov::Node>{param->output(0).get_node(), param->output(0).get_index()});
             }
-            for (const auto& result : model->get_results()) {
-                auto fake_param = std::make_shared<ov::op::v0::Parameter>(result->get_output_element_type(0),
-                                                                          result->get_output_partial_shape(0));
-                const std::string res_name = ov::op::util::create_ie_output_name(result->input_value(0));
+            param->output(0).get_tensor().set_names({param_name});
+            new_param->set_element_type(param->get_element_type());
+            new_param->set_layout(param->get_layout());
+            new_param->output(0).get_rt_info() = param->output(0).get_rt_info();
+            new_param->validate_and_infer_types();
+            m_inputs.emplace_back(
+                ov::Output<const ov::Node>{new_param->output(0).get_node(), param->output(0).get_index()});
```

Comment:
add new nodes here in m_inputs/outputs may introduce other potential issues? like if operate 2.0 public API with legacy 1.0 GPU plugin, this nodes will get duplicated. @ilyachur probably, we need to handle the lifecyle with another proper fix?

Reply:
Another possible fix is to hold `ov::Model` but in this case we also will hold the full original model inside the CompiledModel.
At first sight the creation new inputs/outputs looks good to me. Will take a look

Reply:
@riverlijunjie  port tensors should be reused here:
https://github.com/riverlijunjie/openvino-1/blob/36fedf8bb8aed0236452dd1319723347df7ccaf9/src/inference/src/dev/isync_infer_request.cpp#L266